src/test/java/com/michelin/ns4kafka/service/StreamServiceTest.java

```diff
@@ -337,4 +347,42 @@ void shouldNamespaceBeOwnerOfStreams() {
         assertFalse(streamService.isNamespaceOwnerOfKafkaStream(ns, "test-qua.stream"), "Group ACL missing");
         assertFalse(streamService.isNamespaceOwnerOfKafkaStream(ns, "test-nop.stream"), "No ACL");
     }
+
+    @Test
+    void shouldDeleteKafkaStreamAndRelatedTopics() throws Exception {
+
+        Namespace namespace = Namespace.builder()
+                .metadata(Metadata.builder().name("ns").cluster("local").build())
+                .build();
+        KafkaStream stream = KafkaStream.builder()
+                .metadata(Metadata.builder()
+                        .name("stream")
+                        .namespace("ns")
+                        .cluster("local")
+                        .build())
+                .build();
+        Topic topic1 = Topic.builder()
+                .metadata(Metadata.builder().name("ns.stream-topic1").build())
+                .build();
+
+        Topic topic2 = Topic.builder()
+                .metadata(Metadata.builder().name("ns.stream-topic2").build())
+                .build();
+
+        Topic topic3 = Topic.builder()
+                .metadata(Metadata.builder().name("ns.stream-topic3").build())
+                .build();
+
+        List<Topic> allTopics = List.of(topic1, topic2, topic3);
```

Comment:
We should use different topic names to ensure only the repartition and changelog topic are being deleted:

- abc.stream-topic (not deleted as not ending with -changelog or -repartition)
- abc.stream-topic-repartition (deleted)
- abc.stream-topic-changelog (deleted)
- abc.stream-topic-repartition-test (not deleted as not ending with -changelog or -repartition)
- abc.stream-topic-changelog-test (not deleted as not ending with -changelog or -repartition)
- def.stream-topic-repartition (not deleted as starting by another application.id)
- def.stream-topic-changelog (not deleted as starting by another application.id)